### PR TITLE
Fix bookmark integer overflow and Notes UI bug

### LINKED ISSUE #156: [PE-D] Can add the book with same name and same genre
Because the bookmark does not provide the author, so maybe this will confuse the user. Maybe can add a checking step before accept the input. 

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Lingy12/ped#1

### LINKED ISSUE #162: [PE-D] Cannot display full description for note
It cannot display full description in viewing page

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Lingy12/ped#7

### LINKED ISSUE #163: Integer overflow issues with bookmark
bookmarked page doesn't have regex to check for number of characters, so if user enters a super long string of numbers then there will be integer overflow. We should use BigInteger instead of just int

### PR DESCRIPTION
**Pull request by TanLeYang**
Closes #163  
Added length limit to the regex of the bookmark field to prevent integer overflow.
Wanted to use BigInteger but it would require changing the whole project to use BigInteger so just went with the regex solution.

Closes #162 
Allow text to wrap around instead of being truncated into ellipses when the text is too long to be displayed in a single line.
Here's how it looks now with the maximum number of allowable characters for note and note title:
ss4

DG changes
Modified the DG for the undo/redo feature to include the new changes I added last week

### CODE REVIEW

**Review by pangpuncake — APPROVED** [review]
LGTM!

**Review by angrybunny123 — COMMENTED** [review]
LGTM :)

**Review by pennhanlee — COMMENTED** [review]
LGTM

File: src/main/java/seedu/bookmark/commons/util/StringUtil.java [hunk 1]
```diff
@@ -74,7 +75,7 @@ public static boolean isNonZeroUnsignedInteger(String s) {
      */
     public static String trimLeadingZeroes(String s) {
         requireNonNull(s);
-        int value = Integer.parseInt(s);
+        BigInteger value = new BigInteger(s);
```
Comment (angrybunny123) [inline src/main/java/seedu/bookmark/commons/util/StringUtil.java:78]: I'm not sure if this needs to be here as you mentioned in the PR that you were using the regex solution instead

**Review by angrybunny123 — COMMENTED** [review]
LGTM other than the Big Integer part that I'm not too sure about

**Pull request merged**